File: book_reader/books/book_converter.py
# use $chap$ as an indicator for chapter/sections
# user $title$ as an indocator for the title (title should be on the first chapter)

import glob

chp_code = '$chap$'
ttl_code = '$title$'

files = glob.glob('*.txt')

def add_book(title, chapters):
	return 'books.add(\"%s\", [%s]);\n' % (title, ','.join(chapters))

with open('books.js', 'w') as sink:
	for f in files:
		with open(f, 'r') as source:
			chapters = source.read().split(chp_code)
			title = chapters[0].replace(ttl_code, '').replace('\n', '')
			chapters = [chp_code.join(chapters[:2])] + [chp_code+c for c in chapters[2:]]
			chap_name = []
			# save
			for i,chap in enumerate(chapters):
				with open('%s_%d.js'%(title.replace(' ', '_'), i), 'w') as chap_sink:
					chap_sink.write('var TEXT =\'')					
					for line in chap.split('\n'):
						line = line.replace('\'', '\\\'')
						if not line.replace(' ', ''): continue
						if chp_code in line:
							line = line.replace(chp_code, '')
							chap_name.append('"'+line+'"')
							chap_sink.write('<p class="chapter">%s</p>' % line)
						elif ttl_code in line:
							line = line.replace(ttl_code, '')
							chap_sink.write('<p class="title">%s</p>' % line)
						else:
							chap_sink.write('<p class="text">%s</p>' % line)
					
					chap_sink.write('\';\n')
					chap_sink.write('document.getElementById("book").innerHTML = TEXT;')

			sink.write(add_book(title, chap_name))

File: book_reader/books/book_converter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.IO;
using System.Collections;

namespace converter
{
    class Program
    {
        static string TTLE = "$title$";
        static string CHAP = "$chap$";

        static string CHAPSRT = "var TEXT = \'";
        static string CHAPEND = "\';\ndocument.getElementById(\"book\").innerHTML = TEXT;";

        static string toText(object content, string className)
        {
            return String.Format("<p class=\"{0}\">{1}</p>", className, content.ToString());
        }

        static void Main(string[] args)
        {
            StreamWriter bookWriter = new StreamWriter("books.js");
            StreamWriter chapWriter = null;
            int chapNumber = 0;
            string bookTitle = "";
            ArrayList bookChaps = new ArrayList();
            string[] fileEntries = Directory.GetFiles(@"./");
            foreach (string fileName in fileEntries)
            {
                if (!fileName.EndsWith(".txt"))
                {
                    continue;
                }
                string[] text = File.ReadAllLines(fileName);
                foreach (string l in text)
                {
                    string  line = l.Replace("\\", "\\\\").Replace("\'", "\\\'");
                    // title
                    if (line.StartsWith(TTLE))
                    {
                        if (bookTitle != "")
                        {
                            bookWriter.Write("[\"{0}\"]);\n", String.Join("\",\"",bookChaps.ToArray()));
                            bookChaps.Clear();
                        }
                        bookTitle = line.Remove(0, TTLE.Length);
                        bookWriter.Write("books.add(\"{0}\",", bookTitle);
                        chapNumber = 0;
                    }// chapter 
                    else if (line.StartsWith(CHAP))
                    {
                        if (chapWriter != null)
                        {
                            chapWriter.Write(CHAPEND);
                            chapWriter.Close();
                        }
                        chapWriter = new StreamWriter(String.Format("{0}_{1}.js", bookTitle.Replace(" ", "_"), chapNumber.ToString()));
                        chapNumber++;
                        chapWriter.Write(CHAPSRT);
                        chapWriter.Write(toText(bookTitle, "title"));
                        bookChaps.Add(line.Remove(0, CHAP.Length));
                        chapWriter.Write(toText(bookChaps[bookChaps.Count-1], "chapter"));
                    }
                    else if (chapWriter != null)
                    {
                        chapWriter.Write(toText(line, "text"));
                       
                    }

                }
            }
            bookWriter.Write("[\"{0}\"]);\n", String.Join("\",\"", bookChaps.ToArray()));
            bookWriter.Close();
            chapWriter.Write(CHAPEND);
            chapWriter.Close();
            Console.Out.Write("Done.");
            Console.In.Read();
        }
    }
}
